### PR TITLE
protoc: Change protoc to include generated call option

### PR DESCRIPTION
**Pull request by zasweq**
This PR changes protoc to include generated call option. This will be used in stats handlers. This cannot be merged until gRPC is released with #6926, and we will have to update the minimum version of gRPC required for new versions of protoc-gen-go, and features which use this will need to regenerate their protos (such as Envoy's protos). Per offline discussion, OpenTelemetry e2e testing will be sufficient enough for testing this feature, as protoc-gen-go is implicitly tested in our tests which all use generated code from protoc-gen-go.

RELEASE NOTES:
* protoc: Change protoc to include generated call option

### CODE REVIEW

File: cmd/protoc-gen-go-grpc/grpc.go
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/grpc.go:174]: This needs to be updated to `Version8` now.  The comment just lets users discover more easily when `Version8` became available.

File: cmd/protoc-gen-go-grpc/grpc.go
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/grpc.go:316]: Can be simplified to:

```go
opts := append([]CallOption{grpc.RegisteredMethod()}, opts...) // shadows `opts` so you don't need the subsequent diffs even
``` 

Or use `copy` instead of a loop (and also pre-allocate `len(opts)+1`).

**Review by dfawley — COMMENTED** [review]
LGTM with one nit.  But also blocked on the next gRPC release.

File: cmd/protoc-gen-go-grpc/grpc.go
Comment (zasweq) [reply cmd/protoc-gen-go-grpc/grpc.go:174]: Done.

File: cmd/protoc-gen-go-grpc/grpc.go
Comment (zasweq) [reply cmd/protoc-gen-go-grpc/grpc.go:316]: Changed to former. Chose that since less lines of code and get rid of subsequent diffs.

**Comment by zasweq** [comment]
Finished comments, as discussed offline, let's run this by Eric to see if we even want to go down this route, or if we should preregister on ClientConn.

**Comment by zasweq** [comment]
Also cannot merge until 1.62 release.

**Comment by zasweq** [comment]
Reopened this since I think this is ready to be merged as the release is out.

**Review by dfawley — APPROVED** [review]
Please regenerate all the protos and make sure vet-proto is green before merging.

**Pull request merged**
